website/static/js/mailingListTreebeard.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+var $ = require('jquery');
+var m = require('mithril');
+var Treebeard = require('treebeard');
+var $osf = require('js/osfHelpers');
+var $tb = require('js/treebeardHelpers');
+var projectSettingsTreebeardBase = require('js/projectSettingsTreebeardBase');
+
+function subscribe(item, notification_type, reload) {
+    var id = item.data.node.id;
+    var url = '/api/v1/project/' + id + '/mailing_list/';
```

Comment:
One smaller request: in general, we try to avoid passing hard-coded URLs into knockout code- but more so when we know a refactor/route change is coming.

Ideally the URL for this route would be passed from the backend to JS, rather than hardcoded as a string inside this function.